File: IOT/Anshul-13J_AnshulNigam_2024cs1069_2/2DArrays/SaddlePrice.cpp
#include <iostream>
#include <vector>
using namespace std;

void saddle_point(int n, vector<vector<int>> &arr)
{
    int r[n],c[n],i,min,max,j,flag=0;
    for(i=0;i<n;i++)
    {
        min=arr[i][0]; 
        for(j=0;j<n;j++)
        {
            if(min>arr[i][j])
                min=arr[i][j];
        }
        r[i]=min;
    }
    for(i=0;i<n;i++)
    {
        max=arr[0][i];
        for(j=0;j<n;j++)
        {
            if(max<arr[j][i])
                max=arr[j][i];
        }
        c[i]=max;
    }

    for(i=0;i<n;i++)
    {
        for(j=0;j<n;j++)
        {
            if(r[i]==c[j])
            {
                flag=1;
                cout<<r[i];
                break;
            }
        }
    }
    if(!flag)
        cout<<"Invalid input";
}

int main()
{
    int n;
    cin >> n;
    vector<vector<int>> arr(n, vector<int>(n));
    for (int i = 0; i < n; i++)
        for (int j = 0; j < n; j++)
            cin >> arr[i][j];
    
    saddle_point(n, arr);
}

File: Web/Khushi-Vaish_KhushiVaish_2024cs1147_2/2DArrays/saddlePrice.java
import java.util.*;

class Main {

    public static void main(String[] args) throws Exception {
        Scanner scn = new Scanner(System.in);
        int n = scn.nextInt();
        int[][] arr = new int[n][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                arr[i][j] = scn.nextInt();
            }
        }
        int colno = 0, min = 0;
        for (int i = 0; i < n; i++) {
            min = arr[i][0];
            for (int j = 0; j < n; j++) {
                if (min > arr[i][j]) {
                    min = arr[i][j];
                    colno = j;
                }
            }
            boolean Ans = true;
            for (int r = 0; r < n; r++) {
                if (min < arr[r][colno]) {
                    Ans = false;
                    break;
                }
            }
            if (Ans == true) {
                System.out.print(min);
                return;
            }
        }
        System.out.print("Invalid Input");
    }
}